cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/WidgetCollection/WidgetAttributesCollection/WidgetAttributeInput/AttributeMultipleValuesInput/index.tsx

```diff
@@ -0,0 +1,319 @@
+   * For instance, when the user imports a plugin JSON file into the UI,
+   * It should parse those pre-populated values of widget attributes
+   * and decide what should be 'selectedType' for the component.
+   */
+  function getInitialType() {
+    const widgetAttributeValues = widgetToAttributes[widgetID][field];
+    let newType;
+    if (!widgetAttributeValues || widgetAttributeValues.length === 0) {
+      newType = cleanSupportedTypes[0];
+    } else if (widgetAttributeValues[0].value) {
+      newType = SupportedType.ValueLabelPair;
+    } else if (widgetAttributeValues[0].id) {
+      newType = SupportedType.IDLabelPair;
+    } else {
+      newType = SupportedType.String;
+    }
+    return newType;
+  }
+
+  /*
+   * The input fields can have some pre-populated values.
+   * For instance, when the user imports a plugin JSON file into the UI,
+   * it should parse those pre-populated values of widget attributes
+   * and populate the input fields.
+   */
+  function getCurrentWidgetAttributeValues(newType) {
+    if (!newType) {
+      return '';
+    } else if (newType === SupportedType.Number || newType === SupportedType.String) {
+      return processAttributeValues();
+    } else {
+      return processKeyValueAttributeValues();
+    }
+  }
+
+  /*
+   * Process simple attribute values in order to pass them to the input component.
+   * The component 'CSVWidget' receives the data of following structure.
+   *
+   * Example)
+   *     "GET,POST,PUT,DELETE"
+   */
+  function processAttributeValues() {
+    const widgetAttributeValues = widgetToAttributes[widgetID][field];
+    return widgetAttributeValues ? widgetAttributeValues.join(COMMON_DELIMITER) : '';
+  }
+
+  /*
+   * Process key-value attribute values in order to pass the values to the input component.
+   * The component 'KeyValuePairs' receives the data of following structure.
+   *    { pairs: [{ key: '', value: '' }] }
+   */
+  function processKeyValueAttributeValues() {
+    const widgetAttributeValues = widgetToAttributes[widgetID][field];
+    if (widgetAttributeValues && widgetAttributeValues.length > 0) {
+      return {
+        pairs: widgetAttributeValues.map((keyvalue) => {
+          if (keyvalue.id) {
+            return {
+              key: keyvalue.id,
+              value: keyvalue.label,
+            };
+          } else {
+            return {
+              key: keyvalue.value,
+              value: keyvalue.label,
+            };
+          }
+        }),
+      };
+    } else {
+      return { pairs: [{ key: '', value: '' }] };
+    }
+  }
+
+  const switchInputType = (newType) => {
+    setSelectedType(newType);
+    setCurrentInput(getCurrentWidgetAttributeValues(newType));
+
+    // When user switches the 'selectedType', the value will reset back to an empty array.
+    setWidgetToAttributes((prevObjs) => ({
+      ...prevObjs,
+      [widgetID]: { ...prevObjs[widgetID], [field]: [] },
+    }));
```

Comment:
same comment here regarding setting the state

Reply:
In the later PR, it updates widget attributes as following:
https://github.com/cdapio/cdap/blob/882eb810d3291ed6b39efaf6394a932584abb867/cdap-ui/app/cdap/components/PluginJSONCreator/Create/index.tsx#L121